### PR TITLE
LPS-184945 [Site Initializer] Link Commerce Specification Groups to Labels

### PR DESCRIPTION
**Pull request by joanagrego**

### CODE REVIEW

**Comment by joanagrego** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+	"assetVocabularyName": "Test Asset Vocabulary 1",
+	"currencyCode": "USD",
+	"defaultLanguageId": "en_US",
+	"externalReferenceCode": "TESTCOMMERCECATALOG1",
+	"name": "Test Commerce Catalog 1 update"
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.json:6]: Use capital letter for U

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.products.specifications.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	{
+		"categoryKey": "test-commerce-specification-key-1",
+		"cpDefinitionExternalReferenceCode": "TESTCOMMERCEPRODUCT1",
+		"description": "",
+		"facetable": false,
+		"key": "test-product-specification-1",
+		"productSpecificationValue": {
+			"en_US": "1"
+		},
+		"title": "Test Product Specification 1 update"
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.products.specifications.json:11]: capital letter u

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 3]
```diff
@@ -558,6 +558,29 @@ private void _assertCommerceCatalogs() throws Exception {
...
+				fetchCommerceCatalogByExternalReferenceCode(
+					"TESTCOMMERCECATALOG2", _group.getCompanyId());
+
+		Assert.assertNotNull(commerceCatalog2);
+		Assert.assertEquals(
+			"Test Commerce Catalog 2", commerceCatalog2.getName());
+
+		_assertCPDefinition();
+		_assertCPOption();
+	}
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:582]: please add one more catalog

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 4]
```diff
@@ -722,19 +745,57 @@ private void _assertCommerceNotificationTemplate(
...
+			productSpecificationResource.getProductIdProductSpecificationsPage(
+				cpDefinition1.getCProductId(), Pagination.of(1, 10));
+
+		ProductSpecification productSpecification =
+			productSpecificationPage.fetchFirstItem();
+
+		Assert.assertNotNull(productSpecification);
+		Assert.assertEquals(
+			"test-product-specification-1",
+			productSpecification.getSpecificationKey());
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:782]: move this logic to a method to assert the relation between products and spectifications using localservice

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 5]
```diff
@@ -722,19 +745,57 @@ private void _assertCommerceNotificationTemplate(
...
+	}
+
+	private void _assertCommerceSpecificationProducts2() throws Exception {
+		CPSpecificationOption cpSpecificationOption2 =
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:786]:
```suggestion
		CPSpecificationOption cpSpecificationOption =
```

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 6]
```diff
@@ -558,6 +558,29 @@ private void _assertCommerceCatalogs() throws Exception {
...
+		CommerceCatalog commerceCatalog1 =
+			_commerceCatalogLocalService.
+				fetchCommerceCatalogByExternalReferenceCode(
+					"TESTCOMMERCECATALOG1", _group.getCompanyId());
+
+		Assert.assertNotNull(commerceCatalog1);
+		Assert.assertEquals(
+			"Test Commerce Catalog 1 update", commerceCatalog1.getName());
+
+		CommerceCatalog commerceCatalog2 =
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:571]:
```suggestion
		commerceCatalog =
```

**Review by nNilton — CHANGES_REQUESTED** [review]

**Comment by joanagrego** [comment]
ci:test:sf

**Comment by joanagrego** [comment]
Resent --> [PR](https://github.com/pt-liferay-solutions/liferay-portal/pull/4095)

**Pull request closed without merging**